docs/css-style-guide.md

```diff
@@ -0,0 +1,92 @@
+
+```scss
+// Variables and mixins
+@import 'partials/base';
+
+// Reset default browser styles
+@import 'partials/reset';
+// Basic universal styling for elements (eg. links)
+@import 'partials/elements';
+
+// Components
+@import 'partials/input-field';
+@import 'partials/fancy-modal';
+@import 'partials/a-widget';
+```
+
+ * A 'base' partial which imports all variables and mixins used by other components
+ * A 'variables' partial which defines the colors, typography, grid spacing, animation durations and other foundational elements of the design system
+ * Component partials which define the styles for re-usable components, including the different variations of a component and media queries for responsive layout. These may assume
+   the existence of other variables or mixins but should not import anything themselves.
+
+## Selectors
+
+ * Use simple non-nested class selectors for the majority of styling
+ * Descendant selectors should be avoided. They [are slow](https://developer.mozilla.org/en-US/docs/Web/Guide/CSS/Writing_efficient_CSS#Avoid_the_descendant_selector.21) and more importantly, make it easy to create unintended results due to naming clashes or [specificity conflicts](http://css.maxdesign.com.au/selectutorial/advanced_conflict.htm).
+
+## Naming
+
+ * Use [BEM-style](http://csswizardry.com/2013/01/mindbemding-getting-your-head-round-bem-syntax/) naming for components:
+
+ ```js
+ .component { ... }
+ .component__element { ... }
+ .component__element--modifier { ... }
+ ```
+
+ Where 'component' is a re-usable building block (eg. an input field),
+ 'element' is some part of the component which is not reusable on its own, and
+ '--modifier' represents some static variation of a component (eg.
+ `btn--danger` for a button which performs some destructive action)
+ * Use `is-$state` names for state classes (eg. `is-active`) that are dynamically added or removed from JS code
+ * Use names with a `js-` prefix for classes that are used by JS to get a reference to particular DOM elements. These classes should not appear in stylesheets
+ * Except for tests, only styles with `js-` and `is-` prefixes should be referenced by JavaScript code
+ * Use a `u-` prefix for utility classes. eg. A class for hiding DOM elements would be called `u-hidden`
+
+## SASS Features
+
+ * Use variables and mixins
+ * [Avoid @extends](https://www.sitepoint.com/avoid-sass-extend/). Use mixins instead
+ * Use nesting sparingly
+
+## Colors and Typography
+
+ * Components should use colors and font sizes defined in the 'variables' partial (or one of its include) rather than hard-coding colors
+
+## Layout
+
+ * Most layout should be done with block, inline or flexbox layout
+ * Floats should generally be avoided in favor of flexbox. This avoids the need for clearfix and other hacks. Note that flexbox [may not be appropriate](https://hyp.is/AVKcwo8BvTW_3w8LypJ5/jakearchibald.com/2014/dont-use-flexbox-for-page-layout/) for overall page layout
```

Comment:
Again, I feel like this is saying the same thing twice. The second explanation is clear: "layout within components should usually be done with flexbox rather than with floats" and provides a rationale. It also explains which situations flexbox isn't appropriate in.

So let's just cut the first line?